x/auth/types/codec.go

```diff
@@ -16,12 +22,30 @@ func RegisterCodec(cdc *codec.Codec) {
 	cdc.RegisterConcrete(StdTx{}, "cosmos-sdk/StdTx", nil)
 }
 
-// module wide codec
-var ModuleCdc *codec.Codec
+// RegisterAccountTypeCodec registers an external account type defined in
+// another module for the internal ModuleCdc.
+func RegisterAccountTypeCodec(o interface{}, name string) {
+	ModuleCdc.RegisterConcrete(o, name, nil)
+}
+
+var validAccountTypes = map[string]struct{}{
```

Comment:
See previous comments. This is not necessary/used.

Reply:
I copied it from the gov proposals implementation. Is it unnecessary there as well?

Reply:
If two different modules try and register an account type with the same name (different implementations), will `RegisterConrete` throw? If so, 'validAccountTypes` is redundant.

Reply:
>  Is it unnecessary there as well?

No. It's necessary in x/gov because there are proposal types which are validated against when submitted.

> If two different modules try and register an account type with the same name (different implementations), will RegisterConrete throw? If so, 'validAccountTypes` is redundant.

yes, I believe it will panic